### PR TITLE
Kills shield vest combo

### PR DESCRIPTION
**Pull request by Wayland-Smithy**
## About The Pull Request

Adds checks that *should* prevent anyone from using this silly combo.

## Why It's Good For The Game

It's either this, or deletion. Look on the bright side the sprite is still mostly visible from just one face.

Shield + vest is extra bad for 2 reasons:

1. Blocks the one face of the vest sprite that is easy to see.
2. Adds armor to a build that is supposed to be the definition of a glass cannon.

## Changelog
:cl:
balance: Shield now stops the bomb vest from exploding when equipped and up to 5 seconds after drop.
/:cl:

### CODE REVIEW

**Review by SplinterGP — CHANGES_REQUESTED** [review]
the way this works is bad and you can easily bypass it

File: code/game/objects/items/explosives/bombvest.dm [hunk 1]
```diff
@@ -8,6 +8,23 @@
...
+	RegisterSignal(user, COMSIG_MOB_SHIELD_DROPPED, .proc/shield_dropped)
+
+/obj/item/clothing/suit/storage/marine/harness/boomvest/unequipped(mob/unequipper, slot)
+	. = ..()
+	UnregisterSignal(unequipper, COMSIG_MOB_SHIELD_DROPPED)
+
+///Updates the last shield drop time when one is dropped
+/obj/item/clothing/suit/storage/marine/harness/boomvest/proc/shield_dropped()
+	SIGNAL_HANDLER
+	shield_dropped_last = world.time
```
Comment (TiviPlus) [inline code/game/objects/items/explosives/bombvest.dm:27]: use timer cooldowns

File: code/game/objects/items/explosives/bombvest.dm [hunk 2]
```diff
@@ -18,6 +35,12 @@
 	if(activator.wear_suit != src)
 		to_chat(activator, "Due to the rigging of this device, it can only be detonated while worn.") //If you are going to use this, you have to accept death. No armor allowed.
 		return FALSE
+	if(istype(activator.l_hand, /obj/item/weapon/shield/riot) || istype(activator.r_hand, /obj/item/weapon/shield/riot) || istype(activator.back, /obj/item/weapon/shield/riot))
```
Comment (TiviPlus) [inline code/game/objects/items/explosives/bombvest.dm:38]: why does it check for back though back doesnt protect

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/game/objects/items/explosives/bombvest.dm (hunk 2 above)
Comment (Wayland-Smithy) [reply code/game/objects/items/explosives/bombvest.dm:38]: if you just face north you look like a regular unga?

Comment (Wayland-Smithy) [reply code/game/objects/items/explosives/bombvest.dm:38]: And there are plenty of maps where pushing only north is valid until you run into a xeno that tries to melee you.

**Review by SplinterGP — APPROVED** [review]

**Pull request merged**